exonum/src/blockchain/mod.rs

```diff
@@ -320,18 +320,19 @@ impl Blockchain {
     /// After that invokes `handle_commit` for each service in order of their identifiers
     /// and returns the list of transactions which which were created by the `handle_commit` event.
     #[cfg_attr(feature = "flame_profile", flame)]
-    pub fn commit<'a, I>(
+    pub fn commit<'a, I, S>(
         &mut self,
-        state: &mut State,
+        context: &mut S,
```

Comment:
Another stupid question: why not simply write `&mut ServiceContextMut` here? As far as I understand, references for traits (including mutable ones) work as args and in other contexts because they are constant-size.